main.py

```diff
@@ -66,10 +67,13 @@ def log_starting_thread(thread: Thread):
 
 def get_infinite_iterator(collection: list):
     length = len(collection)
-    i = 0
+    i = (cfg["starting_core"] - 1) * 2
+    if i >= length or i < 0:
+        print("Invalid Core number")
+        quit()
     while True:
         if i >= length:
-            i = 0
+            i = (cfg["starting_core"] - 1) * 2
         yield collection[i]
         i += 1
```

Comment:
i>= length is True every time we reach the end of the thread list. So then we do want to go back to the first thread, regardless of where the user wanted to start the test, right?
Unless you'd want to skip the first few cores altogether for the entire run, but then I would call it different than starting core. "skip_cores" perhaps?